### ISSUE TITLE
Mac shows an icon in the dock when using matplotlib.pyplot.

### ISSUE THREAD

**Issue by luyiming** [matplotlib/matplotlib#12918]
### Bug report

**Bug summary**

Importing `matplotlib.pyplot` shows an icon in the Mac OS dock panel.

**Code for reproduction**

```python
import matplotlib.pyplot as plt
```

**Expected outcome**
No icon in the dock.

**Matplotlib version**

  * Operating system: macOS mojave 10.14
  * Matplotlib version:  3.0.1
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg
  * Python version: 3.6.7
  * Jupyter version (if applicable):
  * Other libraries:

**Comment by luyiming** [comment]
Before:
![Before](https://s1.ax1x.com/2018/12/02/Fua8dU.png)

After:
![After](https://s1.ax1x.com/2018/12/02/Fua3ZT.png)

**Comment by jklymak** [comment]
This is kind of like #12724 

On master if I do 
```
$ python
Python 3.7.0 | packaged by conda-forge | (default, Aug 27 2018, 17:24:52)
[Clang 6.1.0 (clang-602.0.53)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from matplotlib import pyplot as plt
```
I *don't* get a dock icon.

But if I call `ipython` and do the same thing I *do* get a dock icon.  Not sure why ipython behaves differently....

**Comment by WeatherGod** [comment]
ipython injects its own backend when matplotlib is imported

**Comment by jklymak** [comment]
Hmmm, OTOH, if I bisect this, it indeed didn't happen for 3.0.0, so it was likely the backend selection logic we introduced recently.  So, there is some bad ipython interaction there, but maybe we need an ipython expert to get involved.

**Comment by jklymak** [comment]
This still pops an icon on 3.0.x, but is fixed on master, so when 3.1 should fix this annoyance.
